portfolio/src/main/webapp/game_script.js

```diff
@@ -0,0 +1,199 @@
+      ctx = gameArea.context;
+      if (this.type == "text") {
+        ctx.font = this.width + " " + this.height;
+        ctx.fillStyle = color;
+        ctx.fillText(this.text, this.x, this.y);
+      } else {
+        ctx.fillStyle = color;
+        ctx.fillRect(this.x, this.y, this.width, this.height);
+      }
+    }
+    this.newPos = function() {
+        this.x += this.speedX;
+        this.y += this.speedY;
+
+        if (gameArea.x && gameArea.y && gameArea.mouseControl) {
+            /* if mouse position exists and mouse control 
+               is enabled set player y coord to mouse y coord
+               - player only moves vertically*/
+            this.y = gameArea.y;
+        }
+    }
+    this.newSpeed = function() {
+        this.speedX = 0;
+        this.speedY = 0;
+
+        keyPressed = getKey(gameArea.key);
+        switch(keyPressed) {
+            case "UP":
+              this.speedY = -3;
+              break;
+            case "DOWN":
+              this.speedY = 3;
+              break;
+            default:
+              break;
+        } // switch
+    },
+    this.crashWith = function(otherObj) {
+      /* Get left, right, top, bottom for this object */
+      var l = this.x;
+      var r = this.x + (this.width);
+      var top = this.y;
+      var bttm = this.y + (this.height);
+
+      /* Get left, right, top, bottom for other object */
+      var otherL = otherObj.x;
+      var otherR = otherObj.x + (otherObj.width);
+      var otherTop = otherObj.y;
+      var otherBttm = otherObj.y + (otherObj.height);
+      var crash = true;
+
+      var crash = true;
+      if ((bttm < otherTop) ||
+          (top > otherBttm) ||
+          (r < otherL)      ||
+          (l > otherR)) {
+      crash = false;
+    }
+    return crash;
+  }
+} // component
+
+function updateGameArea() {
+  var xObstacle;
+  for (i = 0; i < obstacles.length; i += 1) {
+    if (player.crashWith(obstacles[i])) {
+      gameArea.stop();
+      return;
+    } 
+  }
+  gameArea.clear();
+  /* count frames and add obstacle every 100th frame */
+  gameArea.frameNo += 1;
+  if (gameArea.frameNo == 1 || everyInterval(100)) {
+    /* x coord <- width because new obstacle always spans at end of screen */
+    xObstacle = gameArea.canvas.width;
+
+    minHeight = gameArea.canvas.height/6;
+    maxHeight = gameArea.canvas.height/2;
+    height = Math.floor(Math.random() * (maxHeight - minHeight + 1) + minHeight);
+
+    minGap = 80;
+    maxGap = gameArea.canvas.height/3;
+    gap = gap = Math.floor(Math.random() * (maxGap - minGap + 1) + minGap);
+
+    obstacles.push(new component(50, height, "green", xObstacle, 0));
+    obstacles.push(new component(50, xObstacle - height - gap, "green", xObstacle, height + gap));
+  }
+  for (i = 0; i < obstacles.length; i += 1) {
+    obstacles[i].x += -5;
+    obstacles[i].update();
+  }
+  score.text = "SCORE: " + gameArea.frameNo;
+  score.update();
+  gameArea.updateInstructions();
```

Comment:
If that line is removed, the instructions would no longer show up. This is because the game area (including the instructions) gets cleared on every update (line 160: gameArea.clear() )

Reply:
I had a look at this part of the code and I thought the updateGameArea method is quite verbose so I transferred some of the functionality to methods for better readability. I'll include this change in the future PR, but this can be very easily reverted if it's not a good idea design-wise.